master/internal/rm/kubernetesrm/spec.go

```diff
@@ -488,6 +509,18 @@ func configureUniqueName(t tasks.TaskSpec, rank int) string {
 		t.Description, rank, t.AllocationID, petName.Generate(2, "-"))
 }
 
+func experimentIDTrialIDFromTaskID(taskID string) (experimentID string, trialID string) {
+	res := strings.Split(taskID, ".")
+	experimentID = ""
+	trialID = ""
+	if len(res) == 2 {
+		// expect taskID to be formatted as ExperimentID.TrialRequestID for trials
+		experimentID = res[0]
+		trialID = res[1]
+	}
+	return experimentID, trialID
+}
+
```

Comment:
I don't think parsing the taskID is a good road to go down. I have tried and ran into issues

 https://github.com/determined-ai/determined/commit/108ffea29c6d49151012edda042569368d4f8ecb 
 
 I'm not saying that this will be affected by the same issue just that it is pparsing taskID can cause issues. 
 
Instead can we make a ```map[string]string``` called something like aggegrateLabels on the taskSpec. Then when building the trial task spec we add the experiment ID and trial ID? 
 
 Bradley in the thread was suggesting this
 > We need to keep the task label but having experiment_id and trial_id may still be useful. Now that I think about it, for that it’s probably best if the TaskSpec allows passing down arbitrary labels, and the trial code adds the experiment and trial id, rather than trying to split.

Reply:
Yes, I can go the route Bradley suggested and add a map for these labels on the taskSpec. I didn't go that route originally for a couple reasons:

- The format of taskID is a contract we own; especially if `internal.trialTaskID()` is moved to be accessible by other parts of the code, there shouldn't be an issue with parsing the taskID. (If we use the same function everywhere to form the taskID, then relevant unit tests should fail if it's removed or modified in a way that breaks the parsing.)
- It was tricky to follow how the taskSpec is filled out - it's modified and passed around at many points. I felt that adding another field and ensuring it's populated would be more likely to be error-prone. e.g., it's taking a few tries to get workspace name tracking correct

Reply:
Thanks for explaining your reasoning and answering my (many) questions! It's very helpful and I appreciate it.

I see the elegance of using a map just for pod labels in the task spec. I'm a little concerned that the pod labels can disappear entirely if the map isn't populated for some reason; I think it's better to have a blank label than a missing label. But I also think this is a good solution and we can deal with missing labels if it ever does happen.